File: VirtualGardens/VirtualGardens.Services/Database/Narudzbe.cs
using System;
using System.Collections.Generic;
using VirtualGardens.Services.BaseInterfaces;

namespace VirtualGardens.Services.Database;

public partial class Narudzbe: ISoftDeletable
{
    public int NarudzbaId { get; set; }

    public string BrojNarudzbe { get; set; } = null!;

    public bool? Otkazana { get; set; } = false;

    public DateTime Datum { get; set; } = DateTime.Now;

    public bool Placeno { get; set; } = false;
    public string? StateMachine { get; set; }

    public bool? Status { get; set; }

    public float UkupnaCijena { get; set; }

    public int KorisnikId { get; set; }

    public int? NalogId { get; set; }

    public virtual Korisnici Korisnik { get; set; } = null!;

    public virtual Nalozi? Nalog { get; set; }

    public virtual ICollection<PitanjaOdgovori> PitanjaOdgovoris { get; set; } = new List<PitanjaOdgovori>();

    public virtual ICollection<Setovi> Setovis { get; set; } = new List<Setovi>();
    public bool IsDeleted { get; set; } = false;
    public DateTime? VrijemeBrisanja { get; set; }
}

File: UI/virtualgardens_admin/lib/models/narudzbe.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:virtualgardens_admin/models/korisnici.dart';
import 'package:virtualgardens_admin/models/nalozi.dart';
import 'package:virtualgardens_admin/models/uloga.dart';
import 'package:virtualgardens_admin/models/zaposlenici.dart';

part 'narudzbe.g.dart';

@JsonSerializable()
class Narudzba {
  int narudzbaId;
  String brojNarudzbe;
  bool? otkazana;
  DateTime datum;
  bool placeno;
  bool? status;
  String? stateMachine;
  int ukupnaCijena;
  int korisnikId;
  int? nalogId;
  Korisnik? korisnik;
  Nalog? nalog;

  Narudzba({
    required this.narudzbaId,
    required this.brojNarudzbe,
    this.otkazana,
    required this.datum,
    required this.placeno,
    this.status,
    this.stateMachine,
    required this.ukupnaCijena,
    required this.korisnikId,
    this.nalogId,
    this.korisnik,
    this.nalog,
  });

  factory Narudzba.fromJson(Map<String, dynamic> json) =>
      _$NarudzbaFromJson(json);

  Map<String, dynamic> toJson() => _$NarudzbaToJson(this);
}
